web-services-app/src/main/java/edu/unc/lib/boxc/web/services/processing/DownloadImageService.java

```diff
@@ -30,15 +32,15 @@ public class DownloadImageService {
      * @return a response entity which contains headers and content of the access copy image
      * @throws IOException
      */
-    public ResponseEntity<InputStreamResource> streamImage(String pidString, String size)
+    public ResponseEntity<InputStreamResource> streamImage(String pidString, String size, String filename)
```

Comment:
Just a thought, but if we changed the method signature to 
```
public ResponseEntity<InputStreamResource> streamImage(ContentObjectRecord contentObjectRecord, String size)
```
we could call getFilename directly inside this method since that seems like an internal consideration (and previously was since it used to be responsible for coming up with the filename), and it would have a more consistent API with getSize (using the ContentObjectRecord instead of a string).

Reply:
I do need the `pidString` for the `buildUrl` method but i took your suggestion to move the filename functionality to the service instead of here in the controller.

Reply:
updated the code to figure out pidString fron the ContentObjectRecord